task/sast-coverity-check/0.3/migrations/0.3.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Created for task: sast-coverity-check@0.3
+# Creation time: $(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+declare -r pipeline_file=${1:?missing pipeline file}
+
+
+# Determine the correct image-digest value based on the pipeline file
+if [[ "$pipeline_file" == "docker-build.yaml" ]]; then
+    image_digest_value="\$(tasks.build-image-index.results.IMAGE_DIGEST)"
+else
+    image_digest_value="\$(tasks.build-oci-artifact.results.IMAGE_DIGEST)"
+fi
```

Comment:
A stable way for the purpose of this check is to check if task `build-oci-artifact` exists (my understanding is task `build-oci-artifact` is specific to the `maven-zip-build(-oci-ta)?` pipeline). The value passed to `$1` can be any string.